crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cheated_syscalls.rs

```diff
@@ -65,15 +81,30 @@ pub fn get_block_hash_syscall(
     _remaining_gas: &mut u64,
 ) -> SyscallResult<GetBlockHashResponse> {
 
-    let execution_info_ptr = syscall_handler.get_or_allocate_execution_info_segment(vm)?;
+    // let execution_info_ptr = syscall_handler.get_or_allocate_execution_info_segment(vm)?;
 
-    let cheated_data = cheatnet_state.get_cheated_data(syscall_handler.storage_address());
+    // let cheated_data = cheatnet_state.get_cheated_data(syscall_handler.storage_address());
+
+    // let ptr_cheated_block_info = get_cheated_block_info_ptr(vm, execution_info_ptr, &cheated_data);
+
+    let cheated_block_number = cheatnet_state.block_info.block_number;
+
+    let requested_block_number = request.block_number;
+
+    if cheated_block_number < constants::STORED_BLOCK_HASH_BUFFER
+        || requested_block_number > cheated_block_number - constants::STORED_BLOCK_HASH_BUFFER
+    {
+        let out_of_range_error =
+            Felt::from_hex(BLOCK_NUMBER_OUT_OF_RANGE_ERROR).map_err(SyscallExecutionError::from)?;
+        return Err(SyscallExecutionError::SyscallError { error_data: vec![out_of_range_error] });
+    }
+
+    let key = StorageKey::try_from(Felt::from(requested_block_number))?;
 
-    let ptr_cheated_block_info = get_cheated_block_info_ptr(vm, execution_info_ptr, &cheated_data);
+    let block_hash_contract_address =
+        ContractAddress::try_from(Felt::from(constants::BLOCK_HASH_CONTRACT_ADDRESS))?;
 
-    //TODO: how to get the cheated block number using the ptr_cheated_block_info?
-    //why is the get_cheated_block_info not public?
-    //why is BlockHash not public?
+    //should I get the contractAddress from the cheated block number? how do I do that?
 
     let cheated_block_hash = BlockHash(syscall_handler.state.get_storage_at(block_hash_contract_address, key)?);
 
```

Comment:
Is it correct? Please reconsider it by looking at the blockifier

Reply:
I have reviewed here and I believe it is pretty correct. What did you spot wrong?

Reply:
I need to access the state of the syscall_handler so that I can perform a get storage with the parameters of the contract address of the block hash and the key derived from the requested block number. Seems pretty ok.

Reply:
If there is an error the only possibility is when retrieving the block_hash_contract_address. But why would it be wrong?